src/api/artifact/artifact.router.ts

```diff
@@ -0,0 +1,90 @@
+import express, { NextFunction } from "express";
+import type { Request, Response } from "express";
+import { catchErrors } from "../../utils/middleware";
+import {
+  createArtifact,
+  deleteArtifact,
+  getAllArtifacts,
+  getArtifactById,
+  getArtifactsByCritterId,
+  updateArtifact,
+} from "./artifact.service";
+import {
+  ArtifactCreateBodySchema,
+  ArtifactUpdateBodySchema,
+} from "./artifact.utils";
+import { uuidParamsSchema } from "../../utils/zod_helpers";
+import { apiError } from "../../utils/types";
+import { getCritterById } from "../critter/critter.service";
+
+export const artifactRouter = express.Router();
+
+/**
+ ** Artifact Router Home
+ */
+artifactRouter.get(
+  "/",
+  catchErrors(async (req: Request, res: Response) => {
+    return res.status(200).json(await getAllArtifacts());
+  })
+);
+
+/**
+ ** Create new artifact
+ */
+artifactRouter.post(
+  "/create",
+  catchErrors(async (req: Request, res: Response) => {
+    const artifactData = await ArtifactCreateBodySchema.parse(req.body);
+    const newArtifact = await createArtifact(artifactData);
+    return res.status(201).json(newArtifact);
+  })
+);
+
+/**
+ ** Get artifacts with a critter_id
+ */
+artifactRouter.route("/critter/:id").get(
+  catchErrors(async (req: Request, res: Response) => {
+    // validate uuid and confirm that critter_id exists
+    const { id } = uuidParamsSchema.parse(req.params);
+    await getCritterById(id);
+    const artifacts = await getArtifactsByCritterId(id);
+    if (!artifacts.length) {
```

Comment:
doesnt need to throw error, return [] with valid critter_id. ie: no artifacts found = []